packages/admin-panel-server/src/connections/SessionHandlingApiConnection.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { ApiConnection, AuthHandler, SessionType } from '@tupaia/server-boilerplate';
+import { createBasicHeader } from '@tupaia/utils';
+
+const { MICROSERVICE_CLIENT_USERNAME, MICROSERVICE_CLIENT_SECRET } = process.env;
+const DEFAULT_AUTH_HEADER = createBasicHeader(
+  MICROSERVICE_CLIENT_USERNAME,
+  MICROSERVICE_CLIENT_SECRET,
+);
+
+class SessionSwitchingAuthHandler implements AuthHandler {
+  session?: SessionType;
+
+  constructor(session?: SessionType) {
+    this.session = session;
+  }
+
+  get hasSession() {
+    return !!this.session;
+  }
+
+  async getAuthHeader() {
+    if (this.session) {
+      return this.session.getAuthHeader();
+    }
+
+    return DEFAULT_AUTH_HEADER;
+  }
+}
+
+export abstract class SessionHandlingApiConnection extends ApiConnection {
```

Comment:
I actually don't think you need this whole class/file right? I think for lesmis, because we do allow some public functionalities (users don't need to log in). And my understanding is this SessionHandlingApiConnection class in lesmis-server is to allow that. But admin-panel might not be the same case?

If you agree then MeditrakConnection can just extends ApiConnection from `@tupaia/server-boilerplate` instead of this class!

Reply:
Yes that's a very good point. I'll remove the new class!